apps/language_server/lib/language_server/providers/execute_command/manipulate_pipes.ex

```diff
@@ -0,0 +1,274 @@
+      JsonRpc.send_request("workspace/applyEdit", %{
+        "label" => "Convert function call to pipe operator",
+        "edit" => %{
+          "changes" => %{
+            uri => [%{"range" => edit_range, "newText" => edited_text}]
+          }
+        }
+      })
+
+    case edit_result do
+      {:ok, %{"applied" => true}} ->
+        {:ok, nil}
+
+      other ->
+        {:error, :server_error,
+         "cannot insert spec, workspace/applyEdit returned #{inspect(other)}"}
+    end
+  end
+
+  defp to_pipe_at_cursor(source_file, line, col) do
+    result =
+      ElixirSense.Core.Source.walk_text(
+        source_file.text,
+        %{walked_text: "", function_call: nil, range: nil},
+        fn current_char, remaining_text, current_line, current_col, acc ->
+          if current_line - 1 == line and current_col - 1 == col do
+            {:ok, function_call, call_range} =
+              get_function_call(line, col, acc.walked_text, current_char, remaining_text)
+
+            {remaining_text,
+             %{
+               acc
+               | walked_text: acc.walked_text <> current_char,
+                 function_call: function_call,
+                 range: call_range
+             }}
+          else
+            {remaining_text, %{acc | walked_text: acc.walked_text <> current_char}}
+          end
+        end
+      )
+
+    case result do
+      %{function_call: nil} ->
+        {:error, :function_call_not_found}
+
+      %{function_call: function_call, range: range} ->
+        piped_text = AST.to_pipe(function_call)
+
+        {:ok, %{edited_text: piped_text, edit_range: range}}
+    end
+  end
+
+  defp from_pipe_at_cursor(source_file, line, col) do
+    result =
+      ElixirSense.Core.Source.walk_text(
+        source_file.text,
+        %{walked_text: "", pipe_call: nil, range: nil},
+        fn current_char, remaining_text, current_line, current_col, acc ->
+          if current_line - 1 == line and current_col - 1 == col do
+            {:ok, pipe_call, call_range} =
+              get_pipe_call(line, col, acc.walked_text, current_char, remaining_text)
+
+            {remaining_text,
+             %{
+               acc
+               | walked_text: acc.walked_text <> current_char,
+                 pipe_call: pipe_call,
+                 range: call_range
+             }}
+          else
+            {remaining_text, %{acc | walked_text: acc.walked_text <> current_char}}
+          end
+        end
+      )
+
+    case result do
+      %{pipe_call: nil} ->
+        {:error, :pipe_not_found}
+
+      %{pipe_call: pipe_call, range: range} ->
+        unpiped_text = AST.from_pipe(pipe_call)
+
+        {:ok, %{edited_text: unpiped_text, edit_range: range}}
+    end
+  end
+
+  defp get_function_call(line, col, head, current, original_tail) do
+    tail = do_get_function_call(original_tail, "(", ")")
+
+    {end_line, end_col} =
+      if String.contains?(tail, "\n") do
+        tail_list = String.split(tail, "\n")
```

Comment:
```suggestion
        tail_list = String.split(tail, "\r\n", "\n", "\r")
```

Reply:
Do "\r" characters alone imply newlines? I though they just implied "go back to the first column"

Reply:
nvm, for Mac OS before OS X they do imply newlines